### PR TITLE
[#165763302] Added psp information in transaction detail

### PR DESCRIPTION
**Pull request by laurapugliese**
Added psp information in transaction detail (Only in experimental features)

### CODE REVIEW

**Comment by digitalcitizenship** [comment]
## Affected stories

  * 🌟 [#165763302](https://www.pivotaltracker.com/story/show/165763302): Come CIT voglio vedere nella ricevuta della transazione oltre alla commissione l'eventuale PSP scelto

<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Comment by laurapugliese** [comment]
This PR aims to show the PSP information on the transaction details, if any. The lines 'Payment provider' and 'Payment method' (only in experimental functions) have been added. 
psp_dettaglio_transazione

File: ts/screens/wallet/TransactionDetailsScreen.tsx [hunk 1]
```diff
@@ -185,6 +215,9 @@ class TransactionDetailsScreen extends React.Component<Props> {
       ? this.props.wallets[transaction.idWallet]
       : undefined;
 
+    const transactionPSP =
```
Comment (fpersico) [inline ts/screens/wallet/TransactionDetailsScreen.tsx]: The PSP displayed can be inconsistent.
I complete payment1 using the wallet1 and PSP1.
I complete payment2 using wallet1 but choosing PSP2.
The wallet1 PSP is now PSP2.
If we render the transaction1 the PSP2 will be rendered :(

**Comment by laurapugliese** [comment]
New design:

new_trans_detail

**Comment by laurapugliese** [comment]
> I think you need to change the way psp(s) info are stored in the redux store. Why you are storing a single psp in the store? You need to store all the psps loaded in a psps->byId object. Every you time you go in a transaction details you check if the related psp is in the store otherwise you start fetching it.

@fpersico done.

File: ts/screens/wallet/TransactionDetailsScreen.tsx [hunk 2]
```diff
@@ -256,19 +332,33 @@ class TransactionDetailsScreen extends React.Component<Props> {
...
+  const transaction = ownProps.navigation.getParam("transaction");
+  const idPsp = String(transaction.idPsp);
+
+  const maybePotPspState = fromNullable(pspStateByIdSelector(idPsp)(state));
+  const potPsp = maybePotPspState.map(_ => _.psp).getOrElse(pot.none);
+  const isLoading = pot.isLoading(potPsp);
+
+  return {
+    wallets: pot.toUndefined(getWalletsById(state)),
+    isExperimentalFeaturesEnabled:
```
[the remark this answers is not in the document]
Comment (laurapugliese) [reply ts/screens/wallet/TransactionDetailsScreen.tsx]: @fpersico removed.

**Pull request merged**